golang/cosmos/x/vtransfer/types/baseaddr.go

```diff
@@ -0,0 +1,41 @@
+package types
+
+import (
+	"fmt"
+	"net/url"
+	"strings"
+)
+
+func trimSlashPrefix(s string) string {
+	return strings.TrimPrefix(s, "/")
+}
+
+// ExtractBaseAddress extracts the base address from a parameterized address.
+func ExtractBaseAddress(addr string) (string, error) {
+	parsed, err := url.Parse(addr)
+	if err != nil {
+		return "", err
+	}
+
+	// Remove leading slashes from the path and opaque fields so that only
+	// parsed relative paths match the expected paths.
+	expected := url.URL{
+		Opaque:      trimSlashPrefix(parsed.Opaque),
+		Path:        trimSlashPrefix(parsed.Path),
+		RawPath:     trimSlashPrefix(parsed.RawPath),
+		RawQuery:    parsed.RawQuery,
+		Fragment:    parsed.Fragment,
+		RawFragment: parsed.RawFragment,
+	}
```

Comment:
`url.URL` has [more fields](https://pkg.go.dev/net/url#URL) than being populated here. We're relying on the other fields like `parsed.OmitHost` or `parsed.ForceQuery` to be `false`, which is the default value for the comparison at line 31 to be comparing what we truly care about.

IMO doing
```
if ((parsed.Opaque == trimSlashPrefix(parsed.Opaque) && ...)
```
is still very readable.

Reply:
Yes, I'm relying on all of the non-allowlisted fields to be empty, otherwise to fail.  That was the most concise and robust way of ensuring the parsing worked correctly.  I had originally done an if-ladder and found it to be much less readable than the declaration of `expected` and the single comparison of `parsed` to `expected`.

However, I'll wait for some other opinions on this, and keep it or change it based on the group's agreement.

Reply:
I like this approach and in my inherited Go practice, relying on the zero values for a struct is idiomatic. This is robust against evolution of the struct in a way that an if-ladder would not be, since we want the presence of non-zero values in future fields to result in errors.

Reply:
I agree with @kriskowal. I also rely on Go zero values, I vote for this approach as well